src/dbnode/storage/series/buffer.go

```diff
@@ -114,13 +114,55 @@ type bufferStats struct {
 
 type bufferTickResult struct {
 	mergedOutOfOrderBlocks int
-	evictedBuckets         int
+	evictedBucketTimes     times
+}
+
+// times is a struct that holds an unknown number of times. This is used to
+// to avoid heap allocations as much as possible by trying to not allocate a
+// slice of times. To do this, `timesArraySize` needs to be strategically sized
+// such that for the vast majority of the time, the internal array can hold all
+// the times required so that `slice` is nil.
+type times struct {
+	arrIdx int
+	arr    [timesArraySize]time.Time
+	slice  []time.Time
+}
+
+func (t times) add(newTime time.Time) times {
```

Comment:
more a question: why avoid using a pointer receiver? do you find it affects escape analysis?

Reply:
if so, mind putting in a test to ensure we don't change that behaviour accidentally

Reply:
Double checked with `go build -gcflags='-m'` - it doesn't escape. I'll make it a pointer receiver to have it more usable.